src/components/progress/themes/linear/shared/linear.progress.common.scss

```diff
@@ -14,102 +15,38 @@ $theme: $material;
     color: var-get($theme, 'text-color');
 }
 
-[part~='primary']:not([part~='indeterminate']) {
-    background-color: var-get($theme, 'fill-color-default');
-}
-
-[part~='primary'] {
-    &::after {
-        background-color: var-get($theme, 'fill-color-default');
-    }
-}
-
-[part~='striped'][part~='primary']:not([part~='indeterminate']) {
-    @include stripe-colors(var-get($theme, 'fill-color-default'), var-get($theme, 'stripes-color'), $stripe-size, -45deg);
-}
-
-[part~='striped'][part~='primary'] {
-    &::after {
-        @include stripe-colors(var-get($theme, 'fill-color-default'), var-get($theme, 'stripes-color'), $stripe-size, -45deg);
-    }
-}
-
-[part~='danger']:not([part~='indeterminate']) {
-    background-color: var-get($theme, 'fill-color-danger');
-}
-
-[part~='danger'] {
-    &::after {
-        background-color: var-get($theme, 'fill-color-danger');
-    }
-}
-
-[part~='striped'][part~='danger']:not([part~='indeterminate']) {
-    @include stripe-colors(var-get($theme, 'fill-color-danger'), var-get($theme, 'stripes-color'), $stripe-size, -45deg);
-}
-
-[part~='striped'][part~='danger'] {
-    &::after {
-        @include stripe-colors(var-get($theme, 'fill-color-danger'), var-get($theme, 'stripes-color'), $stripe-size, -45deg);
-    }
-}
-
-[part~='warning']:not([part~='indeterminate']) {
-    background-color: var-get($theme, 'fill-color-warning');
-}
-
-[part~='warning'] {
-    &::after {
-        background-color: var-get($theme, 'fill-color-warning');
-    }
-}
-
-[part~='striped'][part~='warning']:not([part~='indeterminate']) {
-    @include stripe-colors(var-get($theme, 'fill-color-warning'), var-get($theme, 'stripes-color'), $stripe-size, -45deg);
-}
-
-[part~='striped'][part~='warning'] {
-    &::after {
-        @include stripe-colors(var-get($theme, 'fill-color-warning'), var-get($theme, 'stripes-color'), $stripe-size, -45deg);
+// Mixin for base background styles
+@mixin background-styles($part, $color-key) {
+    [part~='#{$part}']:not([part~='indeterminate']) {
+        background-color: var-get($theme, $color-key);
     }
-}
-
-[part~='info']:not([part~='indeterminate']) {
-    background-color: var-get($theme, 'fill-color-info');
-}
-
-[part~='info'] {
-    &::after {
-        background-color: var-get($theme, 'fill-color-info');
+    [part~='#{$part}'] {
+        &::after {
+            background-color: var-get($theme, $color-key);
+        }
     }
 }
 
-[part~='striped'][part~='info']:not([part~='indeterminate']) {
-    @include stripe-colors(var-get($theme, 'fill-color-info'), var-get($theme, 'stripes-color'), $stripe-size, -45deg);
-}
-
-[part~='striped'][part~='info'] {
-    &::after {
-        @include stripe-colors(var-get($theme, 'fill-color-info'), var-get($theme, 'stripes-color'), $stripe-size, -45deg);
+// Mixin for striped styles
+@mixin striped-styles($part, $color-key) {
```

Comment:
This mixin is duplicated in the bootstrap theme, where I commented that having a CSS variable that changes based on the type (info, success, danger, etc.) will be better).